[quant][pyper] Set sparse to False for embedding_bag ops in graph mode

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46003 [quant][pyper] Rename the sparse argument for embedding_bag ops
* **#45997 [quant][pyper] Set sparse to False for embedding_bag ops in graph mode**

Summary:
The current sparse field using in the float module is for sparse gradients, which is not applicable
to inference. The sparse field in the quantizd ops denotes pruned weights.

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24176543](https://our.internmc.facebook.com/intern/diff/D24176543)

supriyar committed:
[quant] Set sparse to False for embedding_bag ops in graph mode
Summary:
The current sparse field using in the float module is for sparse gradients, which is not applicable
to inference. The sparse field in the quantizd ops denotes pruned weights.

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][pyper] Set sparse to False for embedding_bag ops in graph mode"


Summary:
The current sparse field using in the float module is for sparse gradients, which is not applicable
to inference. The sparse field in the quantizd ops denotes pruned weights.

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24176543](https://our.internmc.facebook.com/intern/diff/D24176543)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][pyper] Set sparse to False for embedding_bag ops in graph mode"


Summary:
The current sparse field using in the float module is for sparse gradients, which is not applicable
to inference. The sparse field in the quantizd ops denotes pruned weights.

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24176543](https://our.internmc.facebook.com/intern/diff/D24176543)

[ghstack-poisoned]

vkuzo commented on torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:
```diff
@@ -340,6 +342,8 @@ Node* insertEmbeddingBagOps(Node* observer, const std::string& op_name) {
     for (auto i = 1; i < inputs_size - 1; ++i) {
       qembedding_bag_inputs.push_back(embedding_bag_inputs[i]);
     }
+    // Set the sparse field to 0 for inference.
```
optional: should we mention that this is for sparse gradients (as the pr summary reads), just to make it clear on why?

supriyar committed:
Update on "[quant][pyper] Set sparse to False for embedding_bag ops in graph mode"


Summary:
The current sparse field using in the float module is for sparse gradients, which is not applicable
to inference. The sparse field in the quantizd ops denotes pruned weights.

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24176543](https://our.internmc.facebook.com/intern/diff/D24176543)

[ghstack-poisoned]
